pkg/conditions/conditions.go

```diff
@@ -0,0 +1,229 @@
+func c4g(g Getter) Conditions {
+	return g.GetConditions()
+}
+
+// GetConditions allows the Conditions type to implement the Getter interface.
+func (l Conditions) GetConditions() []metav1.Condition {
+	return l
+}
+
+// Get returns the condition with the given type, otherwise nil is returned.
+func (l Conditions) Get(t string) *metav1.Condition {
+	for _, c := range l {
+		if c.Type == t {
+			return &c
+		}
+	}
+	return nil
+}
+
+// Has returns true if a condition with the given type exists.
+func (l Conditions) Has(t string) bool {
+	return l.Get(t) != nil
+}
+
+// IsTrue returns true if the condition with the given type exists and is
+// True, otherwise false is returned.
+func (l Conditions) IsTrue(t string) bool {
+	if c := l.Get(t); c != nil {
+		return c.Status == metav1.ConditionTrue
+	}
+	return false
+}
+
+// IsFalse returns true if the condition with the given type exists and is
+// False, otherwise false is returned.
+func (l Conditions) IsFalse(t string) bool {
+	if c := l.Get(t); c != nil {
+		return c.Status == metav1.ConditionFalse
+	}
+	return false
+}
+
+// IsUnknown returns true if the condition with the given type exists and is
+// Unknown, otherwise false is returned.
+func (l Conditions) IsUnknown(t string) bool {
+	if c := l.Get(t); c != nil {
+		return c.Status == metav1.ConditionUnknown
+	}
+	return true
+}
+
+// GetReason returns the condition's reason or an empty string if the condition
+// does not exist.
+func (l Conditions) GetReason(t string) string {
+	if c := l.Get(t); c != nil {
+		return c.Reason
+	}
+	return ""
+}
+
+// GetMessage returns the condition's message or an empty string if the
+// condition does not exist.
+func (l Conditions) GetMessage(t string) string {
+	if c := l.Get(t); c != nil {
+		return c.Message
+	}
+	return ""
+}
+
+// GetLastTransitionTime returns the condition's last transition time or a nil
+// value if the condition does not exist.
+func (l Conditions) GetLastTransitionTime(t string) *metav1.Time {
+	if c := l.Get(t); c != nil {
+		return &c.LastTransitionTime
+	}
+	return nil
+}
+
+// Set sets the given condition.
+// If a condition with the same type already exists, its LastTransitionTime is
+// only updated if a change is detected in one of the following fields:
+// Status, Reason, or Message.
+func (l Conditions) Set(c *metav1.Condition) Conditions {
+	if c == nil {
+		return l
+	}
+
+	// Check if the new conditions already exists, and change it only if there
+	// is a status transition (otherwise preserve the current last transition
+	// time).
+	exists := false
+	for i := range l {
+		existingCondition := l[i]
+		if existingCondition.Type == c.Type {
+			exists = true
+			if !hasSameState(&existingCondition, c) {
+				c.LastTransitionTime = metav1.NewTime(
+					time.Now().UTC().Truncate(time.Second))
+				l[i] = *c
+				break
+			}
+			c.LastTransitionTime = existingCondition.LastTransitionTime
+			break
```

Comment:
Is this `break` necessary given it's already end of the for loop?

Reply:
Yes, because it breaks out of the loop.